frameworks/projects/MXRoyale/src/main/royale/mx/messaging/channels/DirectHTTPChannel.as

```diff
@@ -0,0 +1,675 @@
+	 */
+	public class DirectHTTPChannel extends Channel
+	{
+		/**
+		 *  Constructs an instance of a DirectHTTPChannel.
+		 *  The parameters are not used.
+		 *  
+		 *  @langversion 3.0
+		 *  @playerversion Flash 9
+		 *  @playerversion AIR 1.1
+		 *  @productversion BlazeDS 4
+		 *  @productversion LCDS 3 
+		 */
+		public function DirectHTTPChannel(id:String, uri:String="")
+		{
+			super(id, uri);
+			if (uri.length > 0)
+			{
+				var message:String = resourceManager.getString(
+					"messaging", "noURIAllowed");
+				throw new InvalidChannelError(message);
+			}
+			clientId = ("DirectHTTPChannel" + clientCounter++);
+		}
+		
+		/**
+		 * @private
+		 * Used by DirectHTTPMessageResponder to specify a dummy clientId for AcknowledgeMessages.
+		 * Each instance of this channel gets a new clientId.
+		 */ 
+		mx_internal var clientId:String;
+		
+		/**
+		 *  @private
+		 */
+		private var resourceManager:IResourceManager =
+			ResourceManager.getInstance();
+		
+		/**
+		 *  Indicates if this channel is connected.
+		 *  
+		 *  @langversion 3.0
+		 *  @playerversion Flash 9
+		 *  @playerversion AIR 1.1
+		 *  @productversion BlazeDS 4
+		 *  @productversion LCDS 3 
+		 */
+		override public function get connected():Boolean
+		{
+			return true;
+		}
+		
+		/**
+		 *  Indicates the protocol used by this channel.
+		 *  
+		 *  @langversion 3.0
+		 *  @playerversion Flash 9
+		 *  @playerversion AIR 1.1
+		 *  @productversion BlazeDS 4
+		 *  @productversion LCDS 3 
+		 */
+		override public function get protocol():String
+		{
+			return "http";     
+		}
+		
+		//----------------------------------
+		//  realtime
+		//----------------------------------
+		
+		/**
+		 *  @private
+		 *  Returns true if the channel supports realtime behavior via server push or client poll.
+		 */
+		override mx_internal function get realtime():Boolean
+		{
+			return false;
+		}    
+		
+		/**
+		 *  @private
+		 *  Because this channel is always "connected", we ignore any connect timeout
+		 *  that is reported.
+		 */
+		/*COMPILE::SWF
+		override protected function connectTimeoutHandler(event:TimerEvent):void
+		{
+			// Ignore.
+		}
+		
+		COMPILE::JS
+		override protected function connectTimeoutHandler(event:EventDispatcher):void
+		{
+			// Ignore.
+		}*/
+
+		/**
+		 *  Returns the appropriate MessageResponder for the Channel.
+		 *
+		 *  @param agent The MessageAgent sending the message.
+		 * 
+		 *  @param message The IMessage to send.
+		 * 
+		 *  @return The MessageResponder to handle the send result or fault.
+		 *  
+		 *  @langversion 3.0
+		 *  @playerversion Flash 9
+		 *  @playerversion AIR 1.1
+		 *  @productversion BlazeDS 4
+		 *  @productversion LCDS 3 
+		 */
+		COMPILE::SWF
+		override protected function getMessageResponder(agent:MessageAgent, 
+														message:IMessage):MessageResponder
+		{
+			return new DirectHTTPMessageResponder(agent, message, this, new flash.net.URLLoader());
```

Comment:
I think this can be:

return new DirectHTTPMessageResponder(agent, message, this, new URLLoader());

And then you don't need a separate version for COMPILE::JS